src/main/java/com/powsybl/openloadflow/network/PiModelArray.java

```diff
@@ -200,7 +200,7 @@ private Range<Integer> getAllowedPositionRange(AllowedDirection allowedDirection
             case INCREASE:
                 return Range.between(tapPosition - lowTapPosition, models.size());
             case DECREASE:
-                return Range.between(0, tapPosition - lowTapPosition);
+                return Range.between(0, tapPosition - lowTapPosition + 1);
```

Comment:
I agree with this +1 as the maximum of the range is not included in our code. It could be nice to add a unit test that shows the bug.

Reply:
@geofjamg I am not able to add a unit test that shows that this is a bug. Can you just confirm us that this +1 is needed. The PR will be okay for me after your confirmation.